src/js/index.js

```diff
@@ -1,3 +1,7 @@
+/* eslint-disable no-alert */
 import main from './app';
 
-main();
+main().catch((error) => {
+  alert('Server Connection Failed!');
+  console.log(error);
+});
```

Comment:
Error handling does tend to make things messier. I think this is totally fine this early in the project. It can be addressed once the team decides how the behavior should be.